### ISSUE TITLE
Component Documentation?

### ISSUE THREAD

**Issue by chillwinston** [OfficeDev/office-ui-fabric-react#6858]
**Describe the feature that you would like added**
I'm not able to see component level documentation of individual components.  Documentation in general seems a little light, including within the code itself.

For example the "docs" directory for Combobox has a sum total of 8 lines of text across 3 files.  I'm frankly confused as to where the documentation either 

- Is contained
- Or if it truly has only 8 lines of documentation for one component how this project passed internal standards

**What component or utility would this be added to**
All components.

**Have you discussed this feature with our team, and if so, who**
This is the beginning of that discussion I believe?

**Additional context/screenshots**
Much of the code has little to no documentation within the codebase itself.  I was expecting documentation similar to normal Microsoft products and visual components, but am not able to find it thus far.   Am I simply missing where it's located ?

A simple example of the "Button" class in Win forms:
https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.button?view=netframework-4.7.2

There doesn't appear to be any equivalent documentation within UI Fabric and I'm at a loss of how to write clean and accurate code without it.  Am I simply missing where it is ?  At this stage I've looked in the repo and can only see brief paragraphs for Combobox as an example under "docs".

In short:  Confused and have no idea why there's no documentation.

**Comment by cliffkoh** [comment]
Are you aware of the official website at https://developer.microsoft.com/en-us/fabric which has pretty extensive documentation?

**Comment by chillwinston** [comment]
Yes - I did reasonable research before making this feature request.   The core information / documentation I'm after is the React components and their documentation.  The website you reference has a 'components' page here: https://developer.microsoft.com/en-us/fabric#/components.    This has a few short code examples of components but no documentation of the level I reference above for Windows Forms.

Searching for the word "documentation": Google search for 

site:https://developer.microsoft.com/en-us/fabric documentation 

yields 4 pages, none of which reference per-component documentation nor in fact any detailed documentation.

I am literally at a loss as to where the documentation is.  Frankly I have a bad feeling it simply doesn't exist, hence this discussion - it either exists and add a commit to clearly show where it is or it doesn't exist as I suspect it doesn't.  In the 'it doesn't exist' case either it gets added or someone should be asking some hard questions if this is a Microsoft project, especially if Microsoft internally has documentation that we can't publicly see.

At this stage I am literally looking at the source code / test cases for examples of how I might want to use a component - and while I firmly believe in test cases as useful sources of information I truly don't believe that they're in any way a replacement for documentation unless they're structured differently to the Fabric tests that I've already seen.

**Comment by cliffkoh** [comment]
Hmm.. I'm genuinely confused. https://developer.microsoft.com/en-us/fabric#/components/button 

How is this documentation not of the level of the comparable "Button" class in Win forms: https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.button?view=netframework-4.7.2

Note that all the examples come with complete code examples as well, and are interactive.

All properties are documented and there are substantial remarks as well.

**Comment by cliffkoh** [comment]
Could you clarify what is it exactly you find lacking? What is the information you're looking for that is missing?

**Comment by chillwinston** [comment]
Somehow I've managed to miss the contents of the documentation you mention, which actually seems quite good, hence this request can be closed with my apologies.

That being said - I'm used to in-editor support for method signatures and these simply aren't arriving with either VS Code or Visual Studio 2017.  VS Code seems to have better support in general for Fabric. 

The issue relates to the Visual Studio Code click-through documentation being oddly out of sync with the online documentation and indeed the code itself.

Example:
Online documentation https://developer.microsoft.com/en-us/fabric#/components/ComboBox is fine.

However ComboBox.types.d.ts 
- Does not contain the "onChange" member
- Subclass ISelectableDroppableTextProps<T> doesn't have an onChange method
- I'm not sure where onChange is defined on ComboBox with the full onChange definition
- Click through / hover documentation in VS Code is to index.d.ts that has incorrect documentation showing the method as onChange?: FormEventHandler<T>. skipping important index and value
- onChanged is not marked in the code as deprecated but does have index and value

Detail:
Clicking through the onChange attribute of the ComboBox control in VS Code yields the following in index.d.ts :

onChange?: FormEventHandler<T>;

This doesn't include the detail of the call that includes the index and value.  

Hovering the onChange event in VS Code shows the same "blank" / useless method definition as above.

In short : perhaps this issue should be re-raised as support within VS Code for proper code documentation on hover and method / attribute click on the component.  This is the common mechanism for developers to obtain method signatures in C# / Java.  I'm not sure why this is presently the incorrect method signature for TS / VS Code / Visual Studio 2017.

**Comment by cliffkoh** [comment]
Hi @chillwinston, are you using the same version of Fabric that corresponds to the website? If an entire method is missing, you likely are using an older version.

Are you doing development within SPFx?
